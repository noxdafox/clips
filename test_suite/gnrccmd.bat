(defclass CAR (is-a USER) (role concrete))
(make-instance Rolls-Royce of CAR)
(type Rolls-Royce)
(type [Rolls-Royce])
(type (+ 2 2))
(defgeneric describe)
(defmethod describe ((?a INTEGER))
  (if (next-methodp) then
     (bind ?extension (str-cat " " (call-next-method)))
   else
     (bind ?extension ""))
   (str-cat "INTEGER" ?extension))
(describe 3)
(defmethod describe ((?a NUMBER))
   "NUMBER")
(describe 3)
(describe 3.0)
(ppdefgeneric describe)
(ppdefmethod describe 2)
(preview-generic describe 34)
(preview-generic describe 34.0)
(list-defgenerics)
(list-defmethods)
(undefgeneric *)
(list-defgenerics)
(list-defmethods)
(call-next-method 0)
(next-methodp 0)
(preview-generic)
