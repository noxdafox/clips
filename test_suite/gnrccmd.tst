(unwatch all)
(clear)
(conserve-mem off)
(dribble-on "Actual//gnrccmd.out")
(batch "gnrccmd.bat")
(dribble-off)
(clear)
(open "Results//gnrccmd.rsl" gnrccmd "w")
(load "compline.clp")
(printout gnrccmd "gnrccmd.clp differences are as follows:" crlf)
(compare-files "Expected//gnrccmd.out" "Actual//gnrccmd.out" gnrccmd)
(close gnrccmd)
