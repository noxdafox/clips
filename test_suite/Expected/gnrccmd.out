TRUE
CLIPS> (batch "gnrccmd.bat")
TRUE
CLIPS> (defclass CAR (is-a USER) (role concrete))
CLIPS> (make-instance Rolls-Royce of CAR)
[Rolls-Royce]
CLIPS> (type Rolls-Royce)
SYMBOL
CLIPS> (type [Rolls-Royce])
CAR
CLIPS> (type (+ 2 2))
INTEGER
CLIPS> (defgeneric describe)
CLIPS> (defmethod describe ((?a INTEGER))
  (if (next-methodp) then
     (bind ?extension (str-cat " " (call-next-method)))
   else
     (bind ?extension ""))
   (str-cat "INTEGER" ?extension))
CLIPS> (describe 3)
"INTEGER"
CLIPS> (defmethod describe ((?a NUMBER))
   "NUMBER")
CLIPS> (describe 3)
"INTEGER NUMBER"
CLIPS> (describe 3.0)
"NUMBER"
CLIPS> (ppdefgeneric describe)
(defgeneric MAIN::describe)
CLIPS> (ppdefmethod describe 2)
(defmethod MAIN::describe
   ((?a NUMBER))
   "NUMBER")
CLIPS> (preview-generic describe 34)
describe #1  (INTEGER)
describe #2  (NUMBER)
CLIPS> (preview-generic describe 34.0)
describe #2  (NUMBER)
CLIPS> (list-defgenerics)
describe
For a total of 1 defgeneric.
CLIPS> (list-defmethods)
describe #1  (INTEGER)
describe #2  (NUMBER)
For a total of 2 methods.
CLIPS> (undefgeneric *)
CLIPS> (list-defgenerics)
CLIPS> (list-defmethods)
CLIPS> (call-next-method 0)
[ARGACCES4] Function call-next-method expected exactly 0 argument(s)
CLIPS> (next-methodp 0)
[ARGACCES4] Function next-methodp expected exactly 0 argument(s)
CLIPS> (preview-generic)
[ARGACCES4] Function preview-generic expected at least 1 argument(s)
CLIPS> (dribble-off)
